netket/models/equivariant.py

```diff
@@ -24,54 +24,168 @@
+    """
 
+    symmetries: HashableArray
+    """A group of symmetry operations (or array of permutation indices) over which the network should be equivariant.
+    Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
+    """
+    product_table: HashableArray
+    """Product table describing the algebra of the symmetry group
+    Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
     """
+    shape: Tuple
+    """Shape of the translation group"""
+    layers: int
+    """Number of layers (not including sum layer over output)."""
+    features: Tuple
+    """Number of features in each layer starting from the input. If a single number is given,
+    all layers will have the same number of features."""
+    characters: HashableArray
+    """Array specifying the characters of the desired symmetry representation"""
+    dtype: Any = float
+    """The dtype of the weights."""
+    activation: Any = jax.nn.selu
+    """The nonlinear activation function between hidden layers."""
+    output_activation: Any = identity
+    """The nonlinear activation before the output. Defaults to the identity."""
+    imag_part: bool = False
+    """If true return only the imaginary part of the output"""
+    use_bias: bool = True
+    """if True uses a bias in all layers."""
+    precision: Any = None
+    """numerical precision of the computation see `jax.lax.Precision`for details."""
+    kernel_init: NNInitFunc = unit_normal_scaling
+    """Initializer for the Dense layer matrix."""
+    bias_init: NNInitFunc = zeros
+    """Initializer for the hidden bias."""
+
+    def setup(self):
+
+        self.n_symm = np.asarray(self.symmetries).shape[0]
+
+        self.dense_symm = DenseSymmFFT(
+            space_group=self.symmetries,
+            shape=self.shape,
+            features=self.features[0],
+            dtype=self.dtype,
+            use_bias=self.use_bias,
+            kernel_init=self.kernel_init,
+            bias_init=self.bias_init,
+            precision=self.precision,
+        )
+
+        self.equivariant_layers = [
+            DenseEquivariantFFT(
+                product_table=self.product_table,
+                shape=self.shape,
+                in_features=self.features[layer],
+                out_features=self.features[layer + 1],
+                use_bias=self.use_bias,
+                dtype=self.dtype,
+                precision=self.precision,
+                kernel_init=self.kernel_init,
+                bias_init=self.bias_init,
+            )
+            for layer in range(self.layers - 1)
+        ]
+
+    @nn.compact
+    def __call__(self, x):
+        x = self.dense_symm(x)
+
+        for layer in range(self.layers - 1):
+            x = self.activation(x)
+            x = self.equivariant_layers[layer](x)
+
+        x = self.output_activation(x)
+
+        x = logsumexp(
+            x, axis=(1, 2), b=jnp.expand_dims(jnp.asarray(self.characters), (0, 1))
```

Comment:
If it works i'd do it like that, so that it's more general if ever we will work with multibatches...

```suggestion
            x, axis=(-2, -1), b=jnp.expand_dims(jnp.asarray(self.characters), (0, 1))
```